app/login/page.tsx

```diff
@@ -1,81 +1,121 @@
-'use client'
+"use client";
 
-import { Label } from "@/components/ui/label"
-import { Input } from "@/components/ui/input"
-import { Button } from "@/components/ui/button"
-import { SignInOptions, SignInResponse, signIn } from "next-auth/react"
-import { FormEvent, useRef, useState } from "react"
-import { useRouter } from "next/navigation"
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { SignInOptions, SignInResponse, signIn } from "next-auth/react";
+import { FormEvent, useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
 
-const DEFAULT_HOST = 'localhost';
-const DEFAULT_PORT = '6379';
+const DEFAULT_HOST = "localhost";
+const DEFAULT_PORT = "6379";
 
 export default function Page() {
+  const router = useRouter();
+  const [error, setError] = useState(false);
 
-    const router = useRouter()
-    const [error, setError] = useState(false)
+  const [host, setHost] = useState(DEFAULT_HOST);
+  const [port, setPort] = useState(DEFAULT_PORT);
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
 
-    const host = useRef<HTMLInputElement>(null);
-    const port = useRef<HTMLInputElement>(null);
-    const username = useRef<HTMLInputElement>(null);
-    const password = useRef<HTMLInputElement>(null);
+  useEffect(() => {
+    if (typeof window === "undefined") {
+      return;
+    }
 
-    const onSubmit = (e: FormEvent) => {
-        e.preventDefault();
+    const url = new URL(window.location.href);
```

Comment:
@dudizimber there is a more "react" way to get the search params, check other pages.

Reply:
Thre you go